### PR TITLE
fix(chart-testing): Fix helm-chart testing workflow

### PR DESCRIPTION
**Pull request by mjasion**
- Add hidden variable `ci`  which removes annotations and creates docker config secret
- Add hidden variables to pass docker user, password and server

If this fix will work, we will have reliable install and upgrade test process :-)

### CODE REVIEW

File: charts/studio/templates/secret-docker-registry.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{{- if (hasKey .Values "ci") }}
```
Comment (jesper7) [inline charts/studio/templates/secret-docker-registry.yaml:1]: Can we add a comment saying that this file is only used in the CI workflow?
Also, maybe rename the file to `ci-secret-docker-registry.yaml`

File: charts/studio/templates/_helpers.tpl [hunk 2]
```diff
@@ -129,3 +129,7 @@ checksum/secret-studio: {{ include (print $.Template.BasePath "/secret-studio.ya
 {{- define "studio.basePath" -}}
 {{- printf "%s" .Values.global.basePath | trimPrefix "/" | trimSuffix "/" }}
 {{- end }}
+
+{{- define "dockerconfig" -}}
```
Comment (jesper7) [inline charts/studio/templates/_helpers.tpl:133]:
```suggestion
{{- define "ciDockerConfig" -}}
```

**Review by jesper7 — COMMENTED** [review]
Nice, thanks!
Few comments

**Review by jesper7 — COMMENTED** [review]
Small leftover :slightly_smiling_face:

**Review by 0x2b3bfa0 — APPROVED** [review]
Lint is not passing, though

File: charts/studio/templates/ci-secret-docker-registry.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+{{/*
+  The purpose of this file is to create dockerconfigjson secret o pull images during the chart-testing
+  tool execution in the CI. This is needed because the chart-testing tool does not support the use of
+  private registries. This file is only used in the CI, and it is not used in the deployment of the
+  chart.
+*/}}
+{{- if (hasKey .Values "ci") }}
+apiVersion: v1
+data:
+  .dockerconfigjson: {{ include "dockerconfig" . | b64enc | quote }}
```
Comment (mjasion) [inline charts/studio/templates/ci-secret-docker-registry.yaml:10]: It should be renamed too
 
https://github.com/iterative/helm-charts/pull/113/files#diff-1e4f080b0de18f77d739f894999ddc1474b09c387463f24cc2de4e774799d523R133

**Pull request merged**